core/src/main/java/org/arquillian/smart/testing/configuration/ConfigurationLoader.java

```diff
@@ -22,36 +22,60 @@
 
     public static final String SMART_TESTING_YML = "smart-testing.yml";
     public static final String SMART_TESTING_YAML = "smart-testing.yaml";
+    public static final String SMART_TESTING_CONFIG = "smart.testing.config";
+
     private static final Logger logger = Log.getLogger();
 
     public static Configuration load() {
         return load(Paths.get("").toAbsolutePath().toFile());
     }
 
     public static Configuration load(File projectDir) {
+        Map<String, Object> yamlConfiguration = new HashMap<>(0);
+
+        final String customConfigFile = System.getProperty(SMART_TESTING_CONFIG);
+        if (customConfigFile != null) {
+            yamlConfiguration = getConfigurationFromCustomConfigFile(customConfigFile, yamlConfiguration);
+        } else {
+            yamlConfiguration = getConfigurationFromDefaultConfigFile(projectDir, yamlConfiguration);
+        }
+
+        final Object strategiesConfiguration = yamlConfiguration.get("strategiesConfiguration");
+
+        final Configuration configuration = parseConfiguration(yamlConfiguration);
+        if (strategiesConfiguration != null) {
+            configuration.setStrategiesConfig((Map<String, Object>) strategiesConfiguration);
+        }
+
+        return configuration;
+    }
+
+    private static Map<String, Object> getConfigurationFromDefaultConfigFile(File projectDir, Map<String, Object> yamlConfiguration) {
         final File[] files =
             projectDir.listFiles((dir, name) -> name.equals(SMART_TESTING_YML) || name.equals(SMART_TESTING_YAML));
 
         if (files == null) {
             throw new RuntimeException("I/O errors occurs while listing dir " + projectDir);
         }
 
-        Map<String, Object> yamlConfiguration = new HashMap<>(0);
         if (files.length == 0) {
             logger.info("Config file `" + SMART_TESTING_YAML + "` OR `" + SMART_TESTING_YML + "` is not found. "
                 + "Using system properties to load configuration for smart testing.");
         } else {
             yamlConfiguration = getConfigParametersFromFile(getConfigurationFilePath(files));
         }
+        return yamlConfiguration;
+    }
 
-        final Object strategiesConfiguration = yamlConfiguration.get("strategiesConfiguration");
-
-        final Configuration configuration = parseConfiguration(yamlConfiguration);
-        if (strategiesConfiguration != null) {
-            configuration.setStrategiesConfig((Map<String, Object>) strategiesConfiguration);
+    private static Map<String, Object> getConfigurationFromCustomConfigFile(String customConfigFile, Map<String, Object> yamlConfiguration) {
+        final File file = Paths.get(customConfigFile).toAbsolutePath().toFile();
+        if (!file.exists()) {
+            logger.info("Config file `" + file.getName() + "` is not found. "
```

Comment:
if the file doesn't exist, then it should log `WARNING` and should load the configuration using the default way: `getConfigurationFromDefaultConfigFile`

Reply:
I am have a query regarding the Default Configuration File. Currently, there is nothing like a default configuration file that we support. In the event the of missing custom file, we check if the default config file i.e. `smart-testing.yml` is present in the else part. If not, then we use the configuration from the system properties. 
Do you suggest a change to this flow by having some default config file ?

Reply:
> Currently, there is nothing like a default configuration file that we support. 

Yeah, there is no default file, we just have default values for some of the parameters which means that as a result, we can construct a configuration with these default values. 

Correct me if I'm wrong - if the custom file does not exist, you just warn the message and return an empty Map:
```java
        if (!file.exists()) {
            logger.warn("Config file `" + file.getName() + "` is not found. "
                + "Checking for config file `" + SMART_TESTING_YAML + "` OR `" + SMART_TESTING_YML);
        } else {
            return getConfigParametersFromFile(getConfigurationFilePath(file));
        }
        return Collections.EMPTY_MAP;
```
What I meant is that when the custom file does not exist, then we should try to load the configuration from the default locations - as it is done in `getConfigurationFromDefaultConfigFile`